op-monitorism/transaction_monitor/monitor.go

```diff
@@ -0,0 +1,257 @@
+type CheckConfig struct {
+	Type   CheckType              `yaml:"type"`
+	Params map[string]interface{} `yaml:"params"`
+}
+
+// WatchConfig represents the configuration for watching a specific address
+type WatchConfig struct {
+	Address common.Address `yaml:"address"`
+	Filters []CheckConfig  `yaml:"filters"`
+}
+
+type Monitor struct {
+	log          log.Logger
+	client       *ethclient.Client
+	watchConfigs map[common.Address]WatchConfig
+	startBlock   uint64
+	mu           sync.RWMutex
+
+	// metrics
+	transactions        *prometheus.CounterVec
+	unauthorizedTx      *prometheus.CounterVec
+	ethSpent            *prometheus.CounterVec
+	blocksProcessed     *prometheus.CounterVec
+	unexpectedRpcErrors *prometheus.CounterVec
+}
+
+func NewMonitor(ctx context.Context, log log.Logger, m metrics.Factory, cfg CLIConfig) (*Monitor, error) {
+	client, err := ethclient.Dial(cfg.NodeUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial node: %w", err)
+	}
+
+	mon := &Monitor{
+		log:          log,
+		client:       client,
+		watchConfigs: make(map[common.Address]WatchConfig),
+		startBlock:   cfg.StartBlock,
+
+		transactions: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "transactions_total",
+				Help:      "Total number of transactions",
+			},
+			[]string{"from", "to", "status"},
+		),
+
+		unauthorizedTx: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "unauthorized_transactions_total",
+				Help:      "Number of transactions from unauthorized addresses",
+			},
+			[]string{"from"},
+		),
+
+		ethSpent: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "eth_spent_total",
+				Help:      "Cumulative ETH spent by address",
+			},
+			[]string{"address"},
+		),
+
+		blocksProcessed: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "blocks_processed_total",
+				Help:      "Number of blocks processed",
+			},
+			[]string{"number"},
+		),
+
+		unexpectedRpcErrors: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "unexpected_rpc_errors_total",
+				Help:      "Number of unexpected RPC errors",
+			},
+			[]string{"section", "name"},
+		),
+	}
+
+	// Initialize watchConfigs
+	for _, config := range cfg.WatchConfigs {
+		mon.watchConfigs[config.Address] = config
+	}
+
+	return mon, nil
+}
+
+func (m *Monitor) Run(ctx context.Context) {
+	// Initialize starting block
+	currentBlock := m.startBlock
+	if currentBlock == 0 {
+		var err error
+		currentBlock, err = m.client.BlockNumber(ctx)
+		if err != nil {
+			m.log.Error("failed to get initial block number", "err", err)
+			m.unexpectedRpcErrors.WithLabelValues("monitor", "blockNumber").Inc()
+			return
+		}
+		m.log.Info("starting from latest block", "number", currentBlock)
+	}
+
+	ticker := time.NewTicker(time.Second)
```

Comment:
Let's make this interval configurable, also let's probably bump the default to like 12 seconds to match the L1 block time since checking every 1s won't help much